### PR TITLE
Deserialize Response Keys into Object Properties

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Each model class has its properties defined.
Deserialize the response keys into object properties (already done but improve the function a little bit)

Auto completion in IDE
Screen Shot 2023-03-02 at 1 50 15 PM
# Testing

All unit tests pass

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by nwithan8** [comment]
michael-scott-the-office

File: src/models/address.js [hunk 1]
```diff
@@ -5,4 +5,34 @@ import EasyPostObject from './easypost_object';
  * @public
  * @extends EasyPostObject
  */
-export default class Address extends EasyPostObject {}
+export default class Address extends EasyPostObject {
+  static street1;
+
+  static street2;
```
Comment (Justintime50) [inline src/models/address.js:11]: Thought: I hate that our styleguide requires this...

**Review by Justintime50 — COMMENTED** [review]
This looks great, I assume you just wanted a first pass before you added the rest of these?

**Comment by jchen293** [comment]
> This looks great, I assume you just wanted a first pass before you added the rest of these?

Yea, a first pass before I moving forward to the rest

File: src/models/address.js (hunk 1 above)
Comment (jchen293) [reply src/models/address.js]: Per our internal discussion. We will disable `lines-between-class-members` in ESlint for now. We will fix the style guide in another PR.

File: src/services/base_service.js [hunk 2]
```diff
@@ -108,42 +108,35 @@ export default (easypostClient) =>
...
 
-        return mapped;
-        // eslint-disable-next-line no-else-return
-      } else if (typeof response === 'object' && response !== null) {
-        let className;
+      if (typeof response === 'object' && response !== null) {
+        let classObject;
         if (RESOURCES[response.object] !== undefined) {
-          className = RESOURCES[response.object];
+          classObject = new RESOURCES[response.object]();
```
Comment (nwithan8) [inline src/services/base_service.js:122]: *flashbacks to the horrors I've committed in Python...*

**Review by nwithan8 — APPROVED** [review]
Great work Jack!

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: src/models/payload.js [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 import EasyPostObject from './easypost_object';
 
 /**
  * A {@link https://www.easypost.com/docs/api/node#payloads Payload} represents an attempt by EasyPost to send an {@link Event event} to a {@link Webhook webhook}.
  * @public
  * @extends EasyPostObject
  */
-export default class Payload extends EasyPostObject {}
+export default class Payload extends EasyPostObject {
+  static response_body;
```
[the remark this answers is not in the document]
Comment (jchen293) [reply src/models/payload.js:10]: According to our API doc: https://www.easypost.com/docs/api#payload-object. I only need to add `response_body` as EasyPostObject has the rest of the properties.

File: src/models/insurance.js [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+  static from_address;
+  static messages;
+  static provider_id;
+  static provider;
+  static reference;
+  static shipment_id;
+  static status;
+  static to_address;
+  static tracking_code;
+  static trakcer;
```
[the remark this answers is not in the document]
Comment (jchen293) [reply src/models/insurance.js:21]: It was manually typed out

File: src/models/payload.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Justintime50) [reply src/models/payload.js:10]: My bad, everything else is nested under `response_body`, this is fine.

**Review by Justintime50 — APPROVED** [review]
I hope we didn't miss anything

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**